src/NetworkManager.cpp

```diff
@@ -245,6 +296,42 @@ int HttpRequestFuture::Cancel(lua_State *L)
 	return 0;
 }
 
+int WebSocketHandle::Collect(lua_State *L)
+{
+	void *udata = luaL_checkudata(L, 1, "WebSocketHandle");
+	auto handleptr = static_cast<WebSocketHandlePtr*>(udata);
+	handleptr->~shared_ptr();
+	return 0;
+}
+
+int WebSocketHandle::Close(lua_State *L)
+{
+	void *udata = luaL_checkudata(L, 1, "WebSocketHandle");
+	auto handle = *static_cast<WebSocketHandlePtr*>(udata);
+	LUA->YieldLua();
```

Comment:
What does Yield/UnyieldLua (just curious)?

Reply:
It releases the the mutex and allows other threads to use lua. This is necessary to allow for this sequence: webSocket.stop() closes the connection and waits for the sending thread to terminate. The sending thread calls the onMessage callback that wants to run lua code.

Without the yield the program crashes with a mutex deadlock.

Reply:
Makes sense!